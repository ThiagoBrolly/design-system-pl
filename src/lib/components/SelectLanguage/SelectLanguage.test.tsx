import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SelectLanguage } from '.';

describe('<SelectLanguage />', () => {
  it('should correctly render the selectLanguage component', () => {
    const { container } = render(<SelectLanguage locale="pt-BR" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render en-US flag for the selectLanguage by prop', () => {
    render(<SelectLanguage locale="en-US" />);
    expect(
      screen.getByRole('img', {
        name: /selectlanguage country flag_en-us/i,
      }),
    ).toBeInTheDocument();
  });
  it('should render the default img (pt-br) when empty prop is provided', () => {
    render(<SelectLanguage />);
    expect(
      screen.getByRole('img', {
        name: /selectlanguage country flag_pt-br/i,
      }),
    ).toBeInTheDocument();
  });
  it('should render the en-US flag (after clicking/selecting)', () => {
    render(<SelectLanguage />);
    userEvent.click(
      screen.getByRole('img', { name: /selectlanguage country flag_pt-br/i }),
    );
    userEvent.click(
      screen.getByRole('img', { name: /language country flag_en-us/i }),
    );
    expect(
      screen.getByRole('img', {
        name: /selectlanguage country flag_en-us/i,
      }),
    ).toBeInTheDocument();
  });
  it('should render the es-ES flag (after clicking/selecting)', () => {
    render(<SelectLanguage />);
    userEvent.click(
      screen.getByRole('img', { name: /selectlanguage country flag_pt-br/i }),
    );
    userEvent.click(
      screen.getByRole('img', { name: /language country flag_es-es/i }),
    );
    expect(
      screen.getByRole('img', {
        name: /selectlanguage country flag_es-es/i,
      }),
    ).toBeInTheDocument();
  });
});
