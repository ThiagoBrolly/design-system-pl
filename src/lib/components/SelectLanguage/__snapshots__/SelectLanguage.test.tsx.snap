// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectLanguage /> should correctly render the selectLanguage component 1`] = `
.c1 {
  position: relative;
}

.c2 {
  padding: 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c3 {
  width: 4rem;
}

.c4 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
}

.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

@media (max-width:768px) {
  .c3 {
    width: 3rem;
  }
}

@media (max-width:480px) {
  .c3 {
    width: 2.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    id="select-container"
    role="menu"
  >
    <div
      class="c2"
      id="select-dropdwon_header"
    >
      <img
        alt="selectlanguage country flag_pt-BR"
        class="c3"
        id="select-list_image_header"
        src="icon-flag-brazil.svg"
      />
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-label="dropdown-overlay"
    class="c4"
  />
</div>
`;
